### PR TITLE
Refactored DrawingContext and VisualBrush, added DrawingBrush

### PR DESCRIPTION
**Pull request by kekekeks**
1) Removed IVisualBrushRenderer
2) Refactored DrawingContext like this:
image
3)  Changed Skia backend to use SKPicture for visual brushes

### CODE REVIEW

File: src/Avalonia.Base/Media/IVisualBrush.cs [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+using Avalonia.Rendering.Composition.Drawing;
 
 namespace Avalonia.Media
 {
-    /// <summary>
-    /// Paints an area with an <see cref="Visual"/>.
-    /// </summary>
     [NotClientImplementable]
-    public interface IVisualBrush : ITileBrush
+    public interface ISceneBrush : ITileBrush
```
Comment (maxkatz6) [inline src/Avalonia.Base/Media/IVisualBrush.cs:11]: File wasn't renamed.

File: src/Avalonia.Base/Media/DrawingContext.cs [hunk 2]
```diff
@@ -8,96 +8,60 @@
...
-        public void DrawImage(IImage source, Rect sourceRect, Rect destRect, BitmapInterpolationMode bitmapInterpolationMode = default)
+        public virtual void DrawImage(IImage source, Rect sourceRect, Rect destRect,
+            BitmapInterpolationMode bitmapInterpolationMode = default)
         {
             _ = source ?? throw new ArgumentNullException(nameof(source));
-
             source.Draw(this, sourceRect, destRect, bitmapInterpolationMode);
         }
+        
+        public abstract void DrawBitmap(IRef<IBitmapImpl> source, double opacity, Rect sourceRect, Rect destRect, BitmapInterpolationMode bitmapInterpolationMode = BitmapInterpolationMode.Default);
```
Comment (maxkatz6) [inline src/Avalonia.Base/Media/DrawingContext.cs:64]: todo: xml docs

File: src/Avalonia.Base/Media/DrawingGroup.cs [hunk 3]
```diff
@@ -105,7 +102,7 @@ public override Rect GetBounds()
             return rect;
         }
 
-        private class DrawingGroupDrawingContext : IDrawingContextImpl
+        private class DrawingGroupDrawingContext : DrawingContext
```
Comment (maxkatz6) [inline src/Avalonia.Base/Media/DrawingGroup.cs:105]: I think it would make sense to seal specific implementations of DrawingContext.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030399-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030461-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**